app/screens/home/search/results/file_options.tsx

```diff
@@ -0,0 +1,227 @@
+
+const format = 'MMM DD YYYY HH:MM A';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    return {
+        container: {
+            marginTop: -20,
+            marginLeft: -20,
+        },
+        headerContainer: {
+            float: 'left',
+            marginLeft: 20,
+            marginTop: 20,
+            marginBottom: 8,
+        },
+        iconContainer: {
+            marginLeft: -10,
+            marginBottom: 8,
+            alignSelf: 'flex-start',
+        },
+        nameText: {
+            color: theme.centerChannelColor,
+            ...typography('Heading', 400, 'SemiBold'),
+        },
+        infoContainer: {
+            marginVertical: 8,
+            alignItems: 'center',
+            flexDirection: 'row',
+        },
+        infoText: {
+            flexDirection: 'row',
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 200, 'Regular'),
+        },
+        date: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 200, 'Regular'),
+        },
+        menuText: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 200, 'Regular'),
+        },
+    };
+});
+
+type FileOption = {
+    id: string;
+    iconName: string;
+    type: OptionActionType;
+    defaultMessage: string;
+}
+
+const data: FileOption[] = [
+    {
+        id: t('screen.search.results.file_options.download'),
+        iconName: 'download-outline',
+        type: OptionsActions.DOWNLOAD,
+        defaultMessage: 'Download',
+    }, {
+        id: t('screen.search.results.file_options.open_in_channel'),
+        iconName: 'globe',
+        type: OptionsActions.GOTO_CHANNEL,
+        defaultMessage: 'Open in channel',
+    }, {
+        id: t('screen.search.results.file.copy_link'),
+        iconName: 'link-variant',
+        type: OptionsActions.COPY_LINK,
+        defaultMessage: 'Copy Link',
+    },
+];
+
+type Props = {
+    fileInfo: FileInfo;
+}
+const FileOptions = ({fileInfo}: Props) => {
+    const theme = useTheme();
+    const style = getStyleSheet(theme);
+    const isTablet = useIsTablet();
+    const serverUrl = useServerUrl();
+    const [action, setAction] = useState<GalleryAction>('none');
+
+    const galleryItem = {...fileInfo, type: 'image'} as GalleryItemType;
+
+    const handleDownload = useCallback(async () => {
+        setAction('downloading');
+    }, []);
+
+    const handleCopyLink = useCallback(async () => {
+        setAction('copying');
+        dismissBottomSheet();
+    }, []);
+
+    const handleGotoChannel = useCallback(async () => {
+        const fetchedPost = await fetchPostById(serverUrl, fileInfo.post_id, true);
+        const post = fetchedPost.post;
+        const rootId = post?.root_id || post?.id;
+        if (rootId) {
+            fetchAndSwitchToThread(serverUrl, rootId);
+        } else {
+            // what to do?
```

Comment:
The remote action "switchToChannel" (or switchToChannelById, not sure which one is the remote and which one is the local) should be a good starting point. Hopefully that is enough, but some testing would be nice to figure out if the behaviour is the expected.

One potential problem I could imagine is:
- I navigate to the channel from the search results.
- I hit the back button
  - I am back in the search results?
  - or I am back in the channel list?
  - Which one is the expected one?

To get the channel, you can get the id from the post you already fetched.

Reply:
This has changed to opening the permalink view, so the user will see the permalink view and only have the option to close it, or go to the channel from there.